include/CppUTest/UtestMacros.h

```diff
@@ -200,6 +200,29 @@
 #define UNSIGNED_LONGS_EQUAL_LOCATION(expected, actual, text, file, line)\
   { UtestShell::getCurrent()->assertUnsignedLongsEqual((unsigned long)expected, (unsigned long)actual, text, file, line); }
 
+#ifdef CPPUTEST_USE_LONG_LONG
```

Comment:
The `#ifdef` is there so that they'll get a compiler error immediately, instead of a test failure at runtime (which is what would happen otherwise).

The second comment doesn't make sense to me.  There are two cases:

1. If the library was built using AutoMake/CMake, it produces a `config.h` file that "bakes in" the correct setting.   Since applications must always use these header files when using the library (they're part of the devel package), this will Just Work™ and they can't get it wrong unless they're being deliberately silly (in which case it will react as sensibly as it can).  It's never legal for applications to download a separate set of headers when using a pre-built package -- they **must** use the ones from the corresponding devel package.

2. If the application and library are built in VS then it is possible for the two to disagree on the setting.  In this case it makes sense for the user code to fail to compile if they try to use LONGLONG asserts without enabling the feature in their user code (this is entirely independent of whether it's supported in the library or not).  If, after doing so, it turns out that they enabled it in the user code without enabling it in the library, then they'll get a test failure at runtime that tells them this (but this is unlikely as the feature is enabled by default for VS library builds).